ios/src/ImageCropPicker.m

```diff
@@ -400,17 +449,71 @@ - (void)showActivityIndicator:(void (^)(UIActivityIndicatorView*, UIView*))handl
         self.loadingLabel.text = [self.options objectForKey:@"loadingLabelText"];
         [self.loadingLabel setFont:[UIFont boldSystemFontOfSize:18]];
         [loadingView addSubview:self.loadingLabel];
-
+        
         // show all
         [mainView addSubview:loadingView];
         [activityView startAnimating];
-
+        
         handler(activityView, loadingView);
     });
 }
 
 
-- (void) getVideoAsset:(PHAsset*)forAsset completion:(void (^)(NSDictionary* image))completion {   
+
+
+// PS :- Convert Captured Video to the AVAsset
+
+- (void) getVideoAssetFromCamera:(AVURLAsset*)avAsset completion:(void (^)(NSDictionary* image))completion {
+    @try {
+        
+        NSArray *compatiblePresets = [AVAssetExportSession exportPresetsCompatibleWithAsset:avAsset];
+        
+        if ([compatiblePresets containsObject:AVAssetExportPresetLowQuality])
+        {
+            AVAssetExportSession *exportSession = [[AVAssetExportSession alloc]initWithAsset:avAsset presetName:AVAssetExportPresetPassthrough];
+            // save to temp directory
+            NSString *filePath = [[[self getTmpDirectory] stringByAppendingString:[[NSUUID UUID] UUIDString]] stringByAppendingString:@".mp4"];
+            
+            NSString *strFileName = [[[NSUUID UUID] UUIDString] stringByAppendingString:@".mp4"];
+            NSURL *outputURL = [NSURL fileURLWithPath:filePath];
+            
+            exportSession.outputURL = outputURL;
+            NSLog(@"videopath of your mp4 file = %@",filePath);  // PATH OF YOUR .mp4 FILE
+            exportSession.outputFileType = AVFileTypeMPEG4;
+            
+            [exportSession exportAsynchronouslyWithCompletionHandler:^{
+                
+                if ([exportSession status] == AVAssetExportSessionStatusCompleted) {
+                    AVAsset *compressedAsset = [AVAsset assetWithURL:outputURL];
+                    AVAssetTrack *videoTrack = [[compressedAsset tracksWithMediaType:AVMediaTypeVideo] firstObject];
+                    NSNumber *filesize = nil;
+                    [outputURL getResourceValue:&filesize forKey:NSURLFileSizeKey error:nil];
+                    NSDictionary *video = [self createAttachmentResponse:[outputURL absoluteString]
+                                                                withExif:nil
+                                                           withSourceURL:nil
+                                                     withLocalIdentifier:@""
+                                                            withFilename:strFileName
+                                                               withWidth:[NSNumber numberWithFloat:videoTrack.naturalSize.width]
+                                                              withHeight:[NSNumber numberWithFloat:videoTrack.naturalSize.height]
+                                                                withMime:@"video/mp4"
```

Comment:
get mime type from the video.
different iOS versions yield diff types of videos.